pkg/controller/publishingstrategy/publishingstrategy_controller.go

```diff
@@ -100,6 +107,141 @@ func (r *ReconcilePublishingStrategy) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, err
 	}
 
+	// get a list of all ingress on the cluster
+	ingressControllerList := &operatorv1.IngressControllerList{}
+	listOptions := []client.ListOption{
+		client.InNamespace("openshift-ingress-operator"),
+	}
+	err = r.client.List(context.TODO(), ingressControllerList, listOptions...)
+	if err != nil {
+		log.Error(err, "Cannot get list of ingresscontroller")
+		return reconcile.Result{}, err
+	}
+
+	// create temp list of applicationIngress
+	var ingressNotOnClusterList []cloudingressv1alpha1.ApplicationIngress
+	// loop through every applicationingress in publishing strategy and every ingresscontroller in cluster
+	for _, publishingStrategyIngress := range instance.Spec.ApplicationIngress {
+		for _, ingressController := range ingressControllerList.Items {
+			if !isOnCluster(&publishingStrategyIngress, &ingressController) {
+				ingressNotOnClusterList = append(ingressNotOnClusterList, publishingStrategyIngress)
+			}
+		}
+	}
+
+	// unique only
+	ingressList := instance.Spec.ApplicationIngress
+	for _, v := range ingressNotOnClusterList {
+		skip := false
+		for _, u := range ingressList {
+			if v.DNSName == u.DNSName {
+				skip = true
+				break
+			}
+		}
+		if !skip {
+			ingressList = append(ingressList, v)
+		}
+	}
+
+	for _, appingress := range ingressList {
+
+		newCertificate := &corev1.LocalObjectReference{
+			Name: appingress.Certificate.Name,
+		}
+		// default=true
+		if appingress.Default == true {
+			// delete the default appingress on cluster
+			for _, ingresscontroller := range ingressControllerList.Items {
+				if ingresscontroller.Name == defaultIngressName {
+					err := r.client.Delete(context.TODO(), &ingresscontroller)
+					if err != nil {
+						log.Error(err, "failed to delete existing ingresscontroller")
+						return reconcile.Result{}, err
+					}
+				}
+			}
+			log.Info(fmt.Sprintf("in default loop, appingress is: %v", appingress))
```

Comment:
@sam-nguyen7 Was this meant to be deleted with cb8fd99?

Reply:
yes!